backend/src/test/java/gov/cdc/usds/simplereport/api/CSVUploadTest.java

```diff
@@ -0,0 +1,65 @@
+package gov.cdc.usds.simplereport.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.graphql.spring.boot.test.GraphQLResponse;
+import com.graphql.spring.boot.test.GraphQLTestTemplate;
+import gov.cdc.usds.simplereport.test_util.DbTruncator;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.io.InputStream;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.multipart;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Created by nickrobison on 11/21/20
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("dev")
+@AutoConfigureMockMvc
+class CSVUploadTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private GraphQLTestTemplate template;
+
+    @Autowired
+    private DbTruncator _truncator;
+
+    @BeforeEach
+    void setup() {
+        _truncator.truncateAll();
+    }
+
+    @Test
+    void testPersonUpload() throws Exception {
+        try (InputStream inputStream = CSVUploadTest.class.getClassLoader().getResourceAsStream("test-upload.csv")) {
+            assertNotNull(inputStream);
+            byte[] targetArray = new byte[inputStream.available()];
+            final int read = inputStream.read(targetArray);
+            assertTrue(read > 0, "Should have read bytes");
+            mockMvc.perform(post("/upload/person").content(targetArray).contentType("text/csv"))
```

Comment:
@benwarfield-usds Help please! This test is failing with a 415, I'm sure I'm missing something simple regarding the content type headers, but I'm really not sure what it is.

Reply:
I recall going through something like this before. I have a suspicion it was charset-related, but looking at the tests I wrote on that occasion, I don't see anything that suggests that is a thing that actually happened. The only suggestion I have off the top of my head is to convert `targetArray` to a String, and see if that makes the MockMVC successfully negotiate it?

Reply:
@nickrobison-usds - do you have a schema or test file to share? I have the first user list and want to get it in the right format to upload once we have the Prod DB up and running.

Feel free to refer me to the code if that's the best place to see it.

Reply:
It was originally a string, which I thought was the cause of the UTF-8 suffix, now I'm not sure what's going on.